indico_toolkit/auto_populate/populator.py

```diff
@@ -0,0 +1,294 @@
+            base_directory/disclosures/ -> folder containing only disclosures
+            etc. etc.
+        Args:
+            dataset_name (str): Name of created dataset
+            worlflow_name (str): Name of created workflow
+            teach_task_name (str): Name of created teach task
+            labelset_name (str, optional): Name of created labelset. Defaults to None.
+        Returns:
+            Workflow: a Workflow object representation of the newly created workflow
+        """
+        if not labelset_name:
+            labelset_name = f"{teach_task_name}_labelset"
+        self._set_full_doc_classes()
+        # Create empty dataset
+        optional_ocr_options = {
+            "auto_rotate": False,
+            "upscale_images": True,
+            "languages": ["ENG"],
+        }
+        dataset = self.structure.create_dataset(
+            dataset_name=dataset_name,
+            files_to_upload=self.file_paths,
+            read_api=True,
+            single_column=False,
+            **optional_ocr_options,
+        )
+        # Create workflow
+        workflow = self.structure.create_workflow(workflow_name, dataset.id)
+        # Add classifier teach task to workflow (should know labels from parent directory)
+        workflow = self.structure.add_teach_task(
+            task_name=teach_task_name,
+            labelset_name=labelset_name,
+            target_names=list(self.model_classes),
+            dataset_id=dataset.id,
+            workflow_id=workflow.id,
+            model_type="classification",
+        )
+        teach_task_id = workflow.components[-1].model_group.questionnaire_id
+        labelset_id, model_group_id, target_name_map = self._get_teach_task_details(
+            teach_task_id
+        )
+        labels = self._get_classification_labels(model_group_id, target_name_map)
+        self.structure.label_teach_task(
+            label_set_id=labelset_id, labels=labels, model_group_id=model_group_id
+        )
+        return workflow
+
+    def copy_workflow(
+        self, dataset_id: int, teach_task_id: int, workflow_name: str, labelset_id: int = None, data_column: str = None
+    ) -> Workflow:
+        """
+        Create duplicate workflow from dataset and corresponding teach task in same Indico platform.
+        Args:
+            dataset_id (int): The dataset id of the dataset you wish to copy
+            teach_task_id (int): The teach task id of the corresponding teach task to the dataset
+            workflow_name (string): The name of the newly created workflow
+            labelset_id (int, optional): The labelset id of the corresponding labelset to the dataset
+            data_column_id (str, optional): The datacolumn id of the corresponding dataset
+        Returns:
+            Workflow: a Workflow object representation of the newly created workflow
+        """
+        dataset = self.client.call(GetDataset(dataset_id))
+        # get dataset snapshot
+        if not labelset_id:
```

Comment:
I vote to just always get the labelset_id from the dataset object and cut down on lines of code... happy to be overruled if you can think of a good reason to keep option